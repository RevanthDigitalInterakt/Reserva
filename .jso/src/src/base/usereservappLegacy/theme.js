  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.theme = undefined;
  var tokens = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var theme = exports.theme = {
    space: {
      quarck: 4,
      nano: 8,
      micro: 12,
      xxxs: 16,
      xxs: 24,
      xs: 32,
      sm: 40,
      md: 48,
      xl: 64,
      xxl: 80
    },
    shadows: {
      bottomBarShadow: '0px -3px 2px rgba(0,0,0,0.1)',
      topBarShadow: '0px 3px 2px rgba(0,0,0,0.1)'
    },
    colors: {
      preto: tokens.preto,
      vermelhoFechadoRSV: tokens.vermelhoFechadoRSV,
      marinho: tokens.marinho,
      neutroFrio2: tokens.neutroFrio2,
      neutroQuente2: tokens.neutroQuente2,
      vermelhoRSV: tokens.vermelhoRSV,
      verde: tokens.verde,
      neutroQuente1: tokens.neutroQuente1,
      offWhite: tokens.offWhite,
      offWhite15: tokens.offWhite15,
      whiteSecondary: tokens.whiteSecondary,
      neutroFrio1: tokens.neutroFrio1,
      leve: tokens.leve,
      vermelhoAlerta: tokens.vermelhoAlerta,
      verdeSucesso: tokens.verdeSucesso,
      amareloAtencao: tokens.amareloAtencao,
      dropDownBorderColor: tokens.dropDownBorderColor,
      white: tokens.white,
      fullBlack: tokens.fullBlack,
      divider: tokens.divider,
      backgoundInput: tokens.backgoundInput,
      backgroundApp: tokens.backgroundApp,
      transparente: 'transparent',
      modalBackDropColor: tokens.modalBackDropColor,
      borderButton: tokens.borderButton,
      dividerCounter: tokens.dividerCounter,
      backgoundDivider: tokens.backgoundDivider,
      progressTextColor: tokens.progressTextColor,
      backgroundMenuOpened: tokens.backgroundMenuOpened,
      searchBarTextColor: tokens.searchBarTextColor,
      borderColorProduct: tokens.borderColorProduct
    },
    fonts: {
      nunitoBlack: 'Nunito-Black',
      nunitoBlackItalic: 'Nunito-BlackItalic',
      nunitoBold: 'Nunito-Bold',
      nunitoBoldItalic: 'Nunito-BoldItalic',
      nunitoExtraBold: 'Nunito-ExtraBold',
      nunitoExtraBoldItalic: 'Nunito-ExtraBoldItalic',
      nunitoExtraLight: 'Nunito-ExtraLight',
      nunitoExtraLightItalic: 'Nunito-ExtraLightItalic',
      nunitoItalic: 'Nunito-Italic',
      nunitoLight: 'Nunito-Light',
      nunitoLightItalic: 'Nunito-LightItalic',
      nunitoRegular: 'Nunito-Regular',
      nunitoSemiBold: 'Nunito-SemiBold',
      nunitoSemiBoldItalic: 'Nunito-SemiBoldItalic',
      reservaDisplayRegular: 'ReservaDisplay-Regular',
      reservaSerifBlack: 'ReservaSerif-Black',
      reservaSerifBlackItalic: 'ReservaSerif-BlackItalic',
      reservaSerifBold: 'ReservaSerif-Bold',
      reservaSerifBoldItalic: 'ReservaSerif-BoldItalic',
      reservaSerifItalic: 'ReservaSerif-Italic',
      reservaSerifLight: 'ReservaSerif-Light',
      reservaSerifLightItalic: 'ReservaSerif-LightItalic',
      reservaSerifMedium: 'ReservaSerif-Medium',
      reservaSerifMediumItalic: 'ReservaSerif-MediumItalic',
      reservaSerifRegular: 'ReservaSerif-Regular',
      reservaSerifThin: 'ReservaSerif-Thin',
      reservaSerifThinItalic: 'ReservaSerif-ThinItalic',
      reservaSansBlack: 'ReservaSans-Black',
      reservaSansBlackItalic: 'ReservaSans-BlackItalic',
      reservaSansBold: 'ReservaSans-Bold',
      reservaSansBoldItalic: 'ReservaSans-BoldItalic',
      reservaSansItalic: 'ReservaSans-Italic',
      reservaSansLight: 'ReservaSans-Light',
      reservaSansLightItalic: 'ReservaSans-LightItalic',
      reservaSansMedium: 'ReservaSans-Medium',
      reservaSansMediumItalic: 'ReservaSans-MediumItalic',
      reservaSansRegular: 'ReservaSans-Regular',
      reservaSansThin: 'ReservaSans-Thin',
      reservaSansThinItalic: 'ReservaSans-ThinItalic'
    },
    fontWeights: {
      light: 300,
      normal: 400,
      medium: 500,
      SemiBold: 600,
      bold: 700,
      black: 900
    },
    fontSizes: [4, 8, 12, 14, 16, 18, 20, 24, 27, 32, 48],
    radii: {
      pico: 2,
      nano: 5,
      xxxs: 10,
      xs: 15,
      sm: 20,
      pill: 500,
      infinity: 999999
    },
    borderWidths: {
      none: tokens.borderWidthNone,
      hairline: tokens.borderWidthHairline,
      thin: tokens.borderWidthThin,
      thick: tokens.borderWidthThick,
      heavy: tokens.borderWidthHeavy,
      borderDiscountTag: tokens.borderDiscountTag,
      stepper: 3
    }
  };
