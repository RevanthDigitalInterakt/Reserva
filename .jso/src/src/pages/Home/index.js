  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _Home.default;
    }
  });
  var _Home = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
