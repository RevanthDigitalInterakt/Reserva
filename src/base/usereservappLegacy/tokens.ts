export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
