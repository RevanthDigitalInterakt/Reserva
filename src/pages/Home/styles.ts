import { StyleSheet } from 'react-native';
import configDeviceSizes from '../../utils/configDeviceSizes';
import { COLORS } from '../../base/styles';
import { platformType } from '../../utils/platformType';

const styles = StyleSheet.create({
  topBarDefault: {
    zIndex: 999,
    position: 'absolute',
  },
  transparentTopBar: {
    zIndex: 999,
  },
  whiteTopBar: {
    zIndex: 999,
    position: 'absolute',
  },
  rouletWrapper: {
    width: '100%',
    height: '100%',
    backgroundColor: 'transparent',
    position: 'absolute',
    zIndex: 999,
  },
  webView: {
    width: '100%',
    height: configDeviceSizes.DEVICE_HEIGHT * 0.54,
    backgroundColor: 'rgba(0, 0, 0, 0.45)',
    zIndex: 999,
  },
  loaderWrapper: {
    position: 'absolute',
    width: 50,
    height: 50,
    left: configDeviceSizes.DEVICE_WIDTH / 2 - 25,
    top: (configDeviceSizes.DEVICE_HEIGHT * 0.9) / 2 - 25,
    alignItems: 'center',
    justifyContent: 'center',

  },
  closeButton: {
    position: 'absolute',
    left: '10%',
    top: platformType.IOS ? '10%' : 10,
    paddingHorizontal: 10,
    paddingVertical: 5,
    backgroundColor: COLORS.RED,
  },
  closeButtonText: {
    fontSize: 12,
    color: COLORS.WHITE,
  },
});

export default styles;
