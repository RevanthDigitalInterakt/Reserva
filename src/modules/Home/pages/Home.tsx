import React, { useEffect, useLayoutEffect, useState } from 'react';

import { useLazyQuery, useMutation, useQuery } from '@apollo/client';
import AsyncStorage from '@react-native-community/async-storage';
import { useNavigation } from '@react-navigation/native';
import dayjs from 'dayjs';
import timezone from 'dayjs/plugin/timezone';
import utc from 'dayjs/plugin/utc';
import moment from 'moment';
import {
  Dimensions,
  Platform,
  SafeAreaView,
  ScrollView,
  View,
  Linking,
} from 'react-native';
import { FlatList, TouchableHighlight } from 'react-native-gesture-handler';
import { Box, Image } from 'reserva-ui';
import { useAuth } from '../../../context/AuthContext';
import { useCountDown } from '../../../context/ChronometerContext';
import {
  Carrousel,
  CarrouselCard,
  CarrouselTypes,
  configCollection,
  homeQuery,
  HomeQuery,
  ICountDownClock
} from '../../../graphql/homePage/HomeQuery';
import { classicSignInMutation } from '../../../graphql/login/loginMutations';
import { productSearch } from '../../../graphql/products/productSearch';
import { profileQuery } from '../../../graphql/profile/profileQuery';
import { useCheckConnection } from '../../../shared/hooks/useCheckConnection';
import { TopBarDefault } from '../../Menu/components/TopBarDefault';
import { StoreUpdate } from '../../Update/pages/StoreUpdate';
import { Banner } from '../component/Banner';
import { CardsCarrousel } from '../component/CardsCarroussel';
import { DefaultCarrousel } from '../component/Carrousel';
import { DiscoutCodeModal } from '../component/DiscoutCodeModal';
import { CountDownBanner } from '../component/CountDown';
import { Skeleton } from '../component/Skeleton';
import { intervalToDuration } from 'date-fns';
import { useChronometer } from '../../CorreReserva/hooks/useChronometer';

dayjs.extend(utc);
dayjs.extend(timezone);

export const HomeScreen: React.FC<{
  title: string;
}> = () => {
  const navigation = useNavigation();
  const { setEmail, isCookieEmpty, getCredentials, setCookie } = useAuth();
  const { setTime, time } = useCountDown();
  const [modalCodeIsVisible, setModalCodeIsVisible] = useState(true);
  const [getProfile, { data: profileData, loading: profileLoading }] =
    useLazyQuery(profileQuery);
  const [images, setImages] = React.useState<HomeQuery[]>([]);
  const [carrousels, setCarrousels] = React.useState<Carrousel[]>([]);
  const [modalDiscount, setModalDiscount] = React.useState<any>();
  const [countDownClock, setCountDownClock] = React.useState<ICountDownClock>();
  const deviceWidth = Dimensions.get('screen').width;
  const { loading, data, refetch } = useQuery(homeQuery, {
    context: { clientName: 'contentful' },
    variables: { limit: 0 }, // quantidade de itens que iram renderizar
  });
  const { currentValue, start, stop, reset } = useChronometer({
    countDown: true,
    initial: countDownClock?.formattedValue
  });

  const [login, { data: loginData, loading: loginLoading }] = useMutation(
    classicSignInMutation
  );

  const { data: collectionData } = useQuery(configCollection, {
    context: { clientName: 'contentful' },
  });

  const { WithoutInternet } = useCheckConnection({ refetch });

  const { width, height } = Dimensions.get('screen');

  const { data: teste, refetch: refetchTeste } = useQuery(productSearch, {});
  useEffect(() => {
    if (countDownClock) {
      if (new Date(countDownClock?.countdown).getTime() > Date.now()) {
        start();
      }
    }
  }, [countDownClock]);

  useEffect(() => {
    if (currentValue) setTime(currentValue);
  }, [currentValue]);




  useEffect(() => {
    const carrouselsItems: Carrousel[] =
      data?.homePageCollection.items[0].carrouselHomeCollection.items || [];
    setCarrousels(carrouselsItems);

    const arrayImages =
      data?.homePageCollection.items[0].mediasCollection.items.map(
        (imageDescription: any) => ({
          fileName: imageDescription.image.fileName,
          title: imageDescription.image.title,
          width: imageDescription.image.width,
          height: imageDescription.image.height,
          size: imageDescription.image.size,
          url: imageDescription.image.url,
          reference: imageDescription.reference,
          route: imageDescription.route,
        })
      );

    setImages(arrayImages);
  }, [data]);

  useEffect(() => {
    if (collectionData) {
      setModalDiscount(
        collectionData?.configCollection?.items[0].discountCodeBar
      );
      const countDownClock = collectionData?.configCollection?.items[0].countDownClock
      let limitDate
      if (countDownClock?.countdown) {
        limitDate = intervalToDuration({ start: Date.now(), end: new Date(countDownClock?.countdown) });
      }
      if (limitDate) {
        setCountDownClock({
          ...countDownClock,
          formattedValue: `${limitDate?.days * 24 + limitDate.hours}:${limitDate.minutes}:${limitDate.seconds}`
        });
      }
    }
  }, [collectionData]);

  useLayoutEffect(() => {
    if (!isCookieEmpty()) {
      getProfile();
    }
  }, []);

  const loginWithSavedCredentials = async () => {
    const LastLoginAsyncStorageKey = '@RNAuth:lastLogin';

    const lastLogin = await AsyncStorage.getItem(LastLoginAsyncStorageKey);
    const typeLogin = await AsyncStorage.getItem('@RNAuth:typeLogin');
    const nowDate = Date.now();
    const hourToNextLogin = 10;

    if (typeLogin === 'classic') {
      if (nowDate >= Number(lastLogin) + hourToNextLogin * 60 * 60 * 1000) {
        const { email, password } = await getCredentials();
        const { data: loginData, errors } = await login({
          variables: {
            email,
            password,
          },
        });
        if (!loginLoading && loginData?.cookie) {
          await AsyncStorage.setItem('@RNAuth:email', email);
          await AsyncStorage.setItem(
            LastLoginAsyncStorageKey,
            `${moment.now()}`
          );
          await AsyncStorage.setItem('@RNAuth:typeLogin', 'classic');
          await AsyncStorage.setItem('@RNAuth:cookie', loginData.cookie);

          setCookie(loginData.cookie);
          setEmail(email);
        }
      }
    } else if (typeLogin === 'code') {
      if (nowDate >= Number(lastLogin) + 20 * 60 * 60 * 1000) {
        AsyncStorage.removeItem('@RNAuth:cookie');
        AsyncStorage.removeItem('@RNAuth:email');
        AsyncStorage.removeItem('@RNAuth:typeLogin');
        AsyncStorage.removeItem(LastLoginAsyncStorageKey);
        setCookie(null);
        setEmail(null);
      }
    }
  };

  useEffect(() => {
    if (profileData) {
      AsyncStorage.setItem('@RNAuth:email', profileData?.profile?.email);
    } else if (!profileLoading) {
      loginWithSavedCredentials();
    }
  }, [profileData]);

  useEffect(() => {
    // setCookie('asdasdasdasd')
    loginWithSavedCredentials();
    async function getStorage() {
      const wishListData = await AsyncStorage.getItem('@WishList');
    }
    refetchTeste();
    getStorage();
  }, []);
  return (
    <Box flex={1} bg="white">
      <TopBarDefault loading={loading} />
      <StoreUpdate />
      {modalDiscount && (
        <DiscoutCodeModal
          data={modalDiscount}
          isVisible={modalCodeIsVisible}
          onClose={() => {
            setModalCodeIsVisible(false);
          }}
        />
      )}
      <WithoutInternet />
      {loading ? (
        <Skeleton />
      ) : (
        <SafeAreaView
          style={{
            marginBottom: modalCodeIsVisible ? 87 : 50,
          }}
        >
          <ScrollView
            contentContainerStyle={{
              paddingBottom: deviceWidth * 0.02,
            }}
          >
            <Box
              // paddingTop={50}
              style={{
                overflow: 'hidden',
              }}
            >
              {countDownClock && (
                <CountDownBanner countDown={countDownClock} />
              )}
              {carrousels.map((carrousel) => {
                // if (!!carrousel && carrousel.type === CarrouselTypes.mainCarrousel) return <DefaultCarrousel carrousel={carrousel} />
                switch (carrousel?.type) {
                  case CarrouselTypes.mainCarrousel: {
                    return (
                      <>
                        <DefaultCarrousel carrousel={carrousel} />
                      </>
                    )
                    break;
                  }
                  case CarrouselTypes.cardsCarrousel: {
                    const { items } = carrousel.itemsCollection;
                    return items.length > 1 ? (
                      <CardsCarrousel carrousel={carrousel} />
                    ) : (
                      <Banner
                        height={items[0].image.height}
                        reference={items[0].reference}
                        url={items[0].image.url}
                      />
                    );
                    break;
                  }
                  case CarrouselTypes.banner: {
                    const { image, reference } =
                      carrousel.itemsCollection.items[0];
                    return (
                      <Banner
                        height={image.height}
                        reference={reference}
                        url={image.url}
                      />
                    );
                    break;
                  }
                  default: {
                    return <></>;
                    break;
                  }
                }
              })}
            </Box>

            <FlatList
              data={images}
              renderItem={({ item }) => (
                <Banner
                  height={item.height}
                  reference={item.reference}
                  url={item.url}
                  route={item.route}
                />
              )}
              keyExtractor={(item) => item.id}
            />
          </ScrollView>
        </SafeAreaView>
      )}
    </Box>
  );
};
