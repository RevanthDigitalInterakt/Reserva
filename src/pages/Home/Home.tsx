import React, { useCallback, useEffect, useMemo } from 'react';
import {
  ActivityIndicator, Animated, FlatList, SafeAreaView, Text, TouchableOpacity, View,
} from 'react-native';
import utc from 'dayjs/plugin/utc';
import dayjs from 'dayjs';
import timezone from 'dayjs/plugin/timezone';
import { WebView, type WebViewMessageEvent } from 'react-native-webview';

import AsyncStorage from '@react-native-async-storage/async-storage';
import NewBanner from '../../components/Banner/NewBanner';
import { Box } from '../../components/Box/Box';
import ModalSignUpComplete from '../../components/ModalSignUpComplete';
import WithoutInternet from '../../components/WithoutInternet';
import { useConnectivityStore } from '../../zustand/useConnectivityStore';
import { COLORS } from '../../base/styles/colors';
import { useRemoteConfig } from '../../hooks/useRemoteConfig';
import { NewTransparentTopBarDefault } from '../../modules/Menu/components/NewTransparentTopBarDefault';
import { NewWhiteTopBarDefault } from '../../modules/Menu/components/NewWhiteTopBarDefault';
import { TopBarDefault } from '../../modules/Menu/components/TopBarDefault';
import EventProvider from '../../utils/EventProvider';
import { defaultBrand } from '../../utils/defaultWBrand';
import testProps from '../../utils/testProps';
import useAuthModalStore from '../../zustand/useAuthModalStore';
import { useHomeStore } from '../../zustand/useHomeStore';
import HomeCarousels from './components/HomeCarousels';
import HomeCountDown from './components/HomeCountDown';
import HomeDiscountModal from './components/HomeDiscountModal';
import { NewHomeCarousels } from './components/NewHomeCarousels';
import useHomeHeader from './hooks/useHomeHeader';
import styles from './styles';
import { useBagStore } from '../../zustand/useBagStore/useBagStore';
import { ActivityTracking } from '../../components/ActivityTracking';

dayjs.extend(utc);
dayjs.extend(timezone);

function RouletWebview() {
  const { actions, rouletIsOpen, rouletIsLoading } = useBagStore([
    'actions',
    'rouletIsOpen',
    'rouletIsLoading',
  ]);

  const onHandleMessage = useCallback(async (ev: WebViewMessageEvent) => {
    actions.SET_ROULET_LOADING(false);
    if (ev?.nativeEvent?.data) {
      const parsed = JSON.parse(ev?.nativeEvent?.data);
      if (parsed?.data?.reward?.code) {
        actions.SAVE_ROULET_COUPON(
          parsed?.data?.reward?.code,
          parsed?.timestamp,
        );
        await AsyncStorage.setItem('rouletCoupon', JSON.stringify({
          code: parsed?.data?.reward?.code,
          timestamp: parsed?.timestamp,
          blocked: false,
        }));
      }
      if (parsed?.data?.closeMethod === 'button-click') {
        actions.CLOSE_ROULET();
      }
    }
  }, []);

  const webViewUri = useMemo(
    () => `https://www.usereserva.com/files/popconvert.html?${new Date().getTime()}`,
    [],
  );

  return rouletIsOpen ? (
    <View
      style={styles.rouletWrapper}
    >
      <WebView
        source={{
          uri: webViewUri,
        }}
        style={styles.webView}
        cacheEnabled={false}
        cacheMode="LOAD_NO_CACHE"
        onMessage={onHandleMessage}
      />

      <View
        style={[{
          display: rouletIsLoading ? 'flex' : 'none',
        }, styles.loaderWrapper]}
      >
        <ActivityIndicator color={COLORS.RED} size="large" />
      </View>

      <TouchableOpacity
        style={styles.closeButton}
        onPress={() => actions.CLOSE_ROULET()}
      >
        <Text style={styles.closeButtonText}>Fechar</Text>
      </TouchableOpacity>
    </View>
  ) : null;
}

function ListHeader({ newHeaderIsActive }: { newHeaderIsActive: boolean }) {
  return (
    <Box style={{ overflow: 'hidden' }}>
      {newHeaderIsActive ? (
        <NewHomeCarousels />
      ) : (
        <>
          <HomeCountDown />
          <HomeCarousels />
        </>
      )}
    </Box>
  );
}

function Home() {
  const { onLoad, medias, loaded } = useHomeStore([
    'onLoad',
    'medias',
    'loaded',
  ]);
  const { showModalSignUpComplete } = useAuthModalStore([
    'showModalSignUpComplete',
  ]);
  const { isConnected } = useConnectivityStore(['isConnected']);

  const { getBoolean } = useRemoteConfig();
  const newHeaderIsActive = getBoolean('show_new_header');

  const {
    handleScroll,
    topBarDefaultAnimated,
    transparentTopBarAnimated,
    whiteTopBarAnimated,
  } = useHomeHeader();

  const renderHeader = () => (
    <>
      <Animated.View style={[styles.topBarDefault, topBarDefaultAnimated]}>
        <TopBarDefault />
      </Animated.View>
      <Animated.View
        style={[styles.transparentTopBar, transparentTopBarAnimated]}
      >
        <NewTransparentTopBarDefault />
      </Animated.View>
      <Animated.View style={[styles.whiteTopBar, whiteTopBarAnimated]}>
        <NewWhiteTopBarDefault />
      </Animated.View>
    </>
  );

  useEffect(() => {
    if (isConnected && !loaded) {
      onLoad();

      EventProvider.logEvent('page_view', { item_brand: defaultBrand.picapau });
    }
  }, [isConnected, loaded]);

  if (!loaded && !isConnected) {
    return (
      <Box flex={1} bg="white" {...testProps('home_container')}>
        {newHeaderIsActive ? (
          <NewTransparentTopBarDefault />
        ) : (
          <TopBarDefault />
        )}
        <WithoutInternet />
      </Box>
    );
  }

  return (
    <>
      <ActivityTracking />
      <Box flex={1} bg="white" {...testProps('home_container')}>
        {newHeaderIsActive ? renderHeader() : <TopBarDefault />}
        {!newHeaderIsActive ? <HomeDiscountModal /> : null}
        <SafeAreaView {...testProps('home_count_down_container')}>
          <RouletWebview />
          <FlatList
            ListHeaderComponent={
              <ListHeader newHeaderIsActive={newHeaderIsActive} />
                        }
            bounces
            onScroll={handleScroll}
            contentContainerStyle={{ paddingBottom: 100 }}
            keyExtractor={(item) => `home-media-${item.image.url.toString()}-${item.image.title}`}
            data={medias}
            renderItem={({ item }) => (
              <NewBanner
                facets={item.facets}
                image={item.image.url}
                orderBy={item.orderBy}
                reference={item.reference}
                reservaMini={item.reservaMini}
              />
            )}
          />
        </SafeAreaView>
        {!!showModalSignUpComplete && <ModalSignUpComplete />}
      </Box>
    </>
  );
}

export default Home;
