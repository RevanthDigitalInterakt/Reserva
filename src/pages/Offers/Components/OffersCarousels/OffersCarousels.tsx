import React, { useCallback } from 'react';
import { ActivityIndicator, View } from 'react-native';

import { useHomeStore } from '../../../../zustand/useHomeStore';
import { COLORS } from '../../../../base/styles/colors';
import type { OffersCarouselsOutput } from '../../../../base/graphql/generated';
import { HomePageSectionTypeEnum } from '../../../../base/graphql/generated';
import testProps from '../../../../utils/testProps';
import { Box } from '../../../../components/Box/Box';
import OffersMainCarousel from '../OffersMainCarousel/OffersMainCarousel';

export function OffersCarousels() {
  const { offersCarousels, loading } = useHomeStore(['offersCarousels', 'loading']);

  const renderCarousel = useCallback((item: OffersCarouselsOutput) => {
    if (!item.items.length) return null;

    const relationalObject: {
      [key in HomePageSectionTypeEnum | 'DEFAULT']: () => JSX.Element | null;
    } = {
      [HomePageSectionTypeEnum.Main]: () => (
        <OffersMainCarousel data={item} />
      ),
      DEFAULT: () => null,
    };

    return (relationalObject[item.type] || relationalObject.DEFAULT)();
  }, []);

  if (loading) {
    return (
      <Box bg="white" marginY="nano" justifyContent="center">
        <View style={{ height: 100 }} />

        <ActivityIndicator size="small" color={COLORS.BLACK} />
      </Box>
    );
  }

  return (
    <View style={{ height: 400 }}>
      {offersCarousels.map((carousel) => (
        <View
          {...testProps('carousels_cards')}
          key={`item-${carousel.type}-${carousel.showtime}-${carousel.items.length}`}
        >
          {renderCarousel(carousel)}
        </View>
      ))}
    </View>
  );
}
