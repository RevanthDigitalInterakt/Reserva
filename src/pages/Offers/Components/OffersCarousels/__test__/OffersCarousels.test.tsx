import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { MockedProvider } from '@apollo/client/testing';
import { OffersCarousels } from '../OffersCarousels';
import { mockHomeCarouselQuery } from '../../../../Home/__test__/Home.mock';
import { theme } from '../../../../../base/usereservappLegacy/theme';

const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockNavigate }),
}));

jest.mock('react-native-background-timer', () => ({
  stopBackgroundTimer: jest.fn(),
  runBackgroundTimer: jest.fn(),
}));

jest.mock('../../../../../zustand/useHomeStore', () => ({
  useHomeStore: () => ({
    onLoad: () => {},
    offersCarousels: mockHomeCarouselQuery.result.data.homeCarousels,
  }),
}));

const Component = (
  <ThemeProvider theme={theme}>
    <MockedProvider mocks={[mockHomeCarouselQuery]} addTypename={false}>
      <OffersCarousels />
    </MockedProvider>
  </ThemeProvider>
);

describe('OffersCarousel', () => {
  render(Component);

  it('should render properly', () => {
    const card = screen.getByTestId(
      'com.usereserva:id/default_carrousel_container',
    );
    expect(card).toBeOnTheScreen();
  });

  it('should match to snapshot', () => {
    render(Component);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
