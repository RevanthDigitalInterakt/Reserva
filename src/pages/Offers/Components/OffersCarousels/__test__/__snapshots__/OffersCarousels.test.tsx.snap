// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersCarousel should match to snapshot 1`] = `
<View>
  <View
    accessibilityLabel="com.usereserva:id/carousels_cards"
    testID="com.usereserva:id/carousels_cards"
  >
    <View
      accessibilityLabel="com.usereserva:id/default_carrousel_container"
      style={
        {
          "flex": 1,
        }
      }
      testID="com.usereserva:id/default_carrousel_container"
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          collapsable={false}
          handlerTag={2}
          handlerType="PanGestureHandler"
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "height": 400,
                "width": 750,
              },
              {
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "position": "relative",
              },
              {
                "flexDirection": "row",
              },
            ]
          }
          testID="com.usereserva:id/default_carrousel_content"
        >
          <View
            style={
              [
                {
                  "height": 400,
                  "position": "absolute",
                  "width": 750,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                    {
                      "scale": undefined,
                    },
                  ],
                  "zIndex": undefined,
                },
              ]
            }
            testID="__CAROUSEL_ITEM_0_NOT_READY__"
          />
          <View
            style={
              [
                {
                  "height": 400,
                  "position": "absolute",
                  "width": 750,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                    {
                      "scale": undefined,
                    },
                  ],
                  "zIndex": undefined,
                },
              ]
            }
            testID="__CAROUSEL_ITEM_1_NOT_READY__"
          />
          <View
            style={
              [
                {
                  "height": 400,
                  "position": "absolute",
                  "width": 750,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                    {
                      "scale": undefined,
                    },
                  ],
                  "zIndex": undefined,
                },
              ]
            }
            testID="__CAROUSEL_ITEM_2_NOT_READY__"
          />
        </View>
      </View>
      <View
        style={
          [
            {
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "position": "relative",
              "top": 380,
            },
            {
              "width": 73.5,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "borderColor": "#ffffff",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 16.40625,
                "overflow": "hidden",
                "transform": [
                  {
                    "rotateZ": "0deg",
                  },
                ],
                "width": 21.09375,
              },
              {
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 50,
                  "flex": 1,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                  ],
                },
              ]
            }
          />
        </View>
        <View
          style={
            [
              {
                "borderColor": "#ffffff",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 16.40625,
                "overflow": "hidden",
                "transform": [
                  {
                    "rotateZ": "0deg",
                  },
                ],
                "width": 21.09375,
              },
              {
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 50,
                  "flex": 1,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                  ],
                },
              ]
            }
          />
        </View>
        <View
          style={
            [
              {
                "borderColor": "#ffffff",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 16.40625,
                "overflow": "hidden",
                "transform": [
                  {
                    "rotateZ": "0deg",
                  },
                ],
                "width": 21.09375,
              },
              {
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 50,
                  "flex": 1,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                  ],
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="com.usereserva:id/carousels_cards"
    testID="com.usereserva:id/carousels_cards"
  />
  <View
    accessibilityLabel="com.usereserva:id/carousels_cards"
    testID="com.usereserva:id/carousels_cards"
  />
</View>
`;
